chia/wallet/puzzles/dao_proposal_validator.clsp

```diff
@@ -74,11 +74,13 @@
       )
     )
     (c
-      (list CREATE_COIN PROPOSAL_EXCESS_PAYOUT_PUZ_HASH (- coin_amount 1) (list (f (r SINGLETON_STRUCT))))
-      (c
-        (list CREATE_PUZZLE_ANNOUNCEMENT proposal_id)  ; specify the proposal we're talking about
-        conditions
-      )
+      (list CREATE_PUZZLE_ANNOUNCEMENT proposal_id)  ; specify the proposal we're talking about
+      (if (> (- coin_amount 1) 0)
+          (c
+            (list CREATE_COIN PROPOSAL_EXCESS_PAYOUT_PUZ_HASH (- coin_amount 1) (list (f (r SINGLETON_STRUCT))))
+            conditions
+          )
+      conditions)
```

Comment:
two thoughts:
- could we have the wallets ignore 0 coins? this would reduce on-chain code. not urgent but just a thought
- please can you fix the indentation here either way?

Reply:
- the main reasoning for doing it in the chialisp is to prevent proposals creating multiple zero coins which potentially create duplicate condition errors invalidating the whole proposal (particularly for cases where the proposal is spending multiple coins). The other trade off is that while it's more on chain code, it saves the CREATE_COIN cost. I'm not sure which way is more efficient?
- fixed